x/interchainqueries/keeper/keeper.go

```
@@ -306,7 +307,10 @@ func (k Keeper) IterateRegisteredQueries(ctx sdk.Context, fn func(index int64, q
 	i := int64(0)
 	for ; iterator.Valid(); iterator.Next() {
 		query := types.RegisteredQuery{}
-		k.cdc.MustUnmarshal(iterator.Value(), &query)
+		if err := k.cdc.Unmarshal(iterator.Value(), &query); err != nil {
+			k.Logger(ctx).Error("failed to unmarshal registered query %s when iterating: %w", iterator.Key(), err)
+			continue
```

Maybe return an error? Cause logger will not show to a user that error happened.

---

good question. here's my answer: no, there is no reason of doing so and here's why:
- this error happens as the result of an Unmarshal call. this unmarshal call gets a previously registered query data as the argument and tries to wrap the RegisteredQuery struct over it. it's assumed that we've validated the query data before and here we simply reuse it;
- this is an internal loop called by EndBlocker, so there is no way to let the user know about the error because user doesn't call it. instead, the user calls [RegisterInterchainQuery](https://github.com/neutron-org/neutron/blob/feat/add-registered-query-owner/x/interchainqueries/keeper/msg_server.go#L24) which validates the query data passed by the user, and this is the very place where we should check if the user data is correct and return an error any.